Mathlib/MeasureTheory/Measure/Prod.lean

```diff
@@ -691,6 +691,32 @@ protected theorem prod [SFinite μa] [SFinite μc] {f : α → β} {g : γ → 
   have : Measurable (uncurry fun _ : α => g) := hg.1.comp measurable_snd
   hf.skew_product this <| ae_of_all _ fun _ => hg.map_eq
 
+variable (μa μb μc) in
+/-- The measurable equiv induced by the equiv `(α × β) × γ ≃ α × (β × γ)` is measure preserving. -/
+theorem _root_.MeasureTheory.measurePreserving_prodAssoc [SigmaFinite μa] [SigmaFinite μb]
+    [SigmaFinite μc] :
+    MeasurePreserving (MeasurableEquiv.prodAssoc : (α × β) × γ ≃ᵐ α × β × γ)
+      ((μa.prod μb).prod μc) (μa.prod (μb.prod μc)) where
+  measurable := MeasurableEquiv.prodAssoc.measurable
+  map_eq := by
+    refine (FiniteSpanningSetsIn.ext ?_
+      (isPiSystem_measurableSet.prod (isPiSystem_measurableSet.prod isPiSystem_measurableSet))
+      (μa.toFiniteSpanningSetsIn.prod (μb.toFiniteSpanningSetsIn.prod μc.toFiniteSpanningSetsIn))
+        ?_).symm
+    · refine (generateFrom_eq_prod generateFrom_measurableSet
+        (generateFrom_eq_prod ?_ ?_ ?_ ?_) ?_ (IsCountablySpanning.prod ?_ ?_)).symm
+      any_goals exact generateFrom_measurableSet
+      all_goals exact isCountablySpanning_measurableSet
+    · rintro _ ⟨s, _, _, ⟨t, _, ⟨u, _, rfl⟩⟩, rfl⟩
+      rw [MeasurableEquiv.map_apply, MeasurableEquiv.prodAssoc, MeasurableEquiv.coe_mk,
+        Equiv.prod_assoc_preimage, prod_prod, prod_prod, prod_prod, prod_prod, mul_assoc]
```

Comment:
```suggestion
variable (μa μb μc) in
/-- The measurable equiv induced by the equiv `(α × β) × γ ≃ α × (β × γ)` is measure preserving. -/
theorem _root_.MeasureTheory.measurePreserving_prodAssoc [SFinite μa] [SFinite μb]
    [SFinite μc] :
    MeasurePreserving (MeasurableEquiv.prodAssoc : (α × β) × γ ≃ᵐ α × β × γ)
      ((μa.prod μb).prod μc) (μa.prod (μb.prod μc)) where
  measurable := MeasurableEquiv.prodAssoc.measurable
  map_eq := by
    ext s hs
    have A (x : α) : MeasurableSet (Prod.mk x ⁻¹' s) := measurable_prod_mk_left hs
    have B : MeasurableSet (MeasurableEquiv.prodAssoc ⁻¹' s) := 
      MeasurableEquiv.prodAssoc.measurable hs
    simp_rw [map_apply MeasurableEquiv.prodAssoc.measurable hs, prod_apply hs, prod_apply (A _),
      prod_apply B, lintegral_prod _ (measurable_measure_prod_mk_left B).aemeasurable]
    rfl
```
You can weaken the assumptions from `SigmaFinite` to `SFinite` (but for this you need to put it later in the file, because it uses `lintegral_prod`).

Reply:
Indeed (and in fact `lint` tells me that `SFinite μa` is not even needed). Thanks!